packages/chat-headless-react/src/ChatHeadlessProvider.tsx

```diff
@@ -32,7 +32,7 @@ export function ChatHeadlessProvider(
 
   const headless = useMemo(() => {
     const configWithoutSession = { ...config, saveToSessionStorage: false };
-    const headless = new ChatHeadless(updateClientSdk(configWithoutSession));
+    const headless = provideChatHeadless(updateClientSdk(configWithoutSession));
```

Comment:
I believe we want to have ChatHeadlessProvider take in a headless instance like Search (https://github.com/yext/search-headless-react/blob/main/src/SearchHeadlessProvider.tsx). Then, the instance can be internal or public, and also can expose setter functions outside of React, which is what we want to support dynamic context in script tag

Reply:
yeah I have a separate branch where I've made some changes to how headless-react works. Those will be put up separately once this is good.